react-native-hub-app/src/checklist.js

```diff
@@ -0,0 +1,360 @@
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ *
+ * @format
+ * @flow strict-local
+ */
+
+import React from 'react';
+import {FlatList, StyleSheet, View, Text, TouchableOpacity} from 'react-native';
+import {Client, Where} from '@textile/threads-client';
+import {ThreadID} from '@textile/threads-id';
+import {Context, createAPISig} from '@textile/textile';
+import { Libp2pCryptoIdentity } from '@textile/threads-core';
+
+const MAX_STEPS = 2;
+const sleep = (m) => new Promise((r) => setTimeout(r, m));
+
+class CheckList extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  // you could also do this, so no constructor needed
+  state = {
+    test: 0,
+    tests: [
+      {key: 'Test 0', name: 'Create signature', status: 0},
+      {key: 'Test 1', name: 'Create user token', status: 0},
+      {key: 'Test 2', name: 'Create new client DB', status: 0},
+      {key: 'Test 3', name: 'pending', status: 0},
+      {key: 'Test 4', name: 'pending', status: 0},
+    ],
+    errorMessage: '',
+  };
+
+  incrementStatus(index) {
+    const tests = this.state.tests;
+    const data = tests[index];
+    if (data.status >= MAX_STEPS) {
+      return;
+    }
+    data.status = data.status += 1;
+    tests[index] = data;
+    this.setState({
+      tests: tests,
+    });
+  }
+
+  async test(testNumber) {
+    const tests = this.state.tests;
+    const data = tests[testNumber];
+    try {
+      let {ctx, db, threadId} = this.state;
+      switch (testNumber) {
+        case 0: {
+          ctx = new Context();
+          const sig = await createAPISig('secret');
+          ctx = ctx.withAPIKey('brlbyskqsntf4w52jhhvrvbdzuq').withAPISig(sig);
+
+          data.status = 2;
+          tests[testNumber] = data;
+          this.setState({
+            ctx: ctx,
+            tests: tests,
+          });
+
+          break;
+        }
+        case 1: {
+          db = new Client(ctx);
+          const identity = await Libp2pCryptoIdentity.fromRandom();
+          const tok = await db.getToken(identity);
+          ctx = ctx.withToken(tok);
+
+          // Update our app state with success
+          data.status = 2;
+          tests[testNumber] = data;
+          this.setState({
+            identity: identity,
+            tok: tok,
+            ctx: ctx,
+            db: db,
+            tests: tests,
+          });
+          break;
+        }
+        case 2: {
+          ctx = ctx.withThreadName('foo');
+          threadId = ThreadID.fromRandom();
+          // // Update our app state with success
+          await db.newDB(threadId, ctx);
```

Comment:
Possible to debug or otherwise console.log the ctx object here? That error (Token required) is surprising given you've already updated the context by this point.